HRIS.Services/Services/AuthService.cs

```diff
@@ -45,13 +49,13 @@ private static bool IsTokenExpired(string token)
                 return _cachedAccessToken;
             }
         }
-        var tokenEndpoint = $"{EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}oauth/token";
+        var tokenEndpoint = $"{_authManagement.Issuer ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}oauth/token";
         var requestBody = new Dictionary<string, string>
         {
-            { "client_id",  EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID},
-            { "client_secret", EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_SECRET },
+            { "client_id",  _authManagement.ClientId ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID},
+            { "client_secret", _authManagement.ClientSecret ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_SECRET },
             { "grant_type", "client_credentials" },
-            { "audience", $"{EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}api/v2/" }
+            { "audience", $"{_authManagement.Issuer ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}api/v2/" }
```

Comment:
Refactor: can this entire thing perhaps be a variable set once in the constructor?

string _clientId = _authManagement.ClientId ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID;

Same applies to rest of code.